cmd/podman/images/list.go

```diff
@@ -189,23 +197,27 @@ func sortImages(imageS []*entities.ImageSummary) []imageReporter {
 	}
 
 	sort.Slice(imgs, sortFunc(listFlag.sort, imgs))
-	return imgs
+	return imgs, err
 }
 
-func tokenRepoTag(tag string) (string, string) {
-	tokens := strings.Split(tag, ":")
-	switch len(tokens) {
-	case 0:
-		return tag, ""
-	case 1:
-		return tokens[0], ""
-	case 2:
-		return tokens[0], tokens[1]
-	case 3:
-		return tokens[0] + ":" + tokens[1], tokens[2]
-	default:
-		return "<N/A>", ""
+func tokenRepoTag(ref string) (string, string, error) {
+
+	repo, err := reference.Parse(ref)
+	if err != nil {
+		return "<N/A>", "", err
 	}
+	named, ok := repo.(reference.Named)
+	if ok == false {
+		return "<N/A>", "", errors.New("Repository parsing failure")
```

Comment:
I think we should be a bit more forgiving and return `ref, "", nil`

Reply:
Also note that errors (by convention) should always be in lower caps. This way, they can still be wrapped. Only logs should start with a capital letter to make them easier to brain parse.